libraries/standard/http/src/http_client.c

```diff
@@ -930,18 +946,102 @@ HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    HTTPParsingContext_t parsingContext;
+
+    memset( &parsingContext, 0, sizeof( HTTPParsingContext_t ) );
+
+    if( pResponse == NULL )
+    {
+        IotLogError( "Parameter check failed: pResponse is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pResponse->pBuffer == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders->pBuffer is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pResponse->bufferLen == 0u )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders->bufferLen is 0: Buffer len should be > 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pHeaderName == NULL )
+    {
+        IotLogError( "Parameter check failed: Input header name is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( headerNameLen == 0u )
+    {
+        IotLogError( "Parameter check failed: Input header name length is 0: pHeaderName should be > 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pHeaderValueLoc == NULL )
+    {
+        IotLogError( "Parameter check failed: Output parameter for header value location is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pHeaderValueLen == NULL )
+    {
+        IotLogError( "Parameter check failed: Output parameter for header value length is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else
+    {
+        /* Empty else for MISRA 15.7 compliance. */
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        /* Initialize Parsing context with our callback. */
+        returnStatus = HTTPClient_InitializeParsingContext( &parsingContext, NULL );
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        returnStatus = HTTPClient_FindHeaderInResponse( &parsingContext,
+                                                        pResponse->pBuffer,
+                                                        pResponse->bufferLen,
+                                                        pHeaderName,
+                                                        headerNameLen,
+                                                        pHeaderValueLoc,
+                                                        pHeaderValueLen );
+
+        if( returnStatus == HTTP_SUCCESS )
+        {
+            /* Header value found present in buffer. */
+            IotLogDebugWithArgs( "Found requested header in response: "
+                                 "HeaderName=%.*s, ValueLoc=%.*s",
+                                 headerNameLen, pHeaderName,
+                                 *pHeaderValueLen, *pHeaderValueLoc );
+        }
+        else if( returnStatus == HTTP_HEADER_NOT_FOUND )
+        {
+            /* Header is not present in buffer. */
+            IotLogWarnWithArgs( "Unable to read header from response: "
```

Comment:
Would removing "Unable to read header from response: " make the message any less obvious?

Reply:
Wanted to separate out the aspects of <Outcome> and <Failure> in the message, but in this case, the lines are blurred between them. I will remove them in the interest of code size.